### ISSUE TITLE
SC2123 detection is unreliable

### ISSUE THREAD

**Issue by ale5000-git** [koalaman/shellcheck#2739]
#### For bugs
- Rule Id (if any, e.g. SC1000):  SC2123
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/usr/bin/env sh

_pathsep=':'
PATH="mypath${_pathsep:?}${PATH}"
PATH="/mypath${_pathsep:?}${PATH}"

```

#### Here's what shellcheck currently says:

One line give a warning while the other no.

#### Here's what I wanted or expected to see:

It would be nice if it could detect that PATH is not overwritten in either of two.
